app/services/current_user_provider.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+class CurrentUserProvider
+  include CurrentOauth
+
+  CURRENT_USER_KEY ||= '_CURRENT_USER_KEY'
+
+  attr_accessor :request, :response
+
+  class << self
+    def new_instance(request, response: nil)
+      ins = new(request.env, response)
+      ins.request = request
+      ins
+    end
+  end
+
+  def initialize(env, response)
+    @env = env
+    @response = response
+  end
+
+  def current_user
+    return @env[CURRENT_USER_KEY] if @env.key?(CURRENT_USER_KEY)
+
+    current_user = nil
+    current_user = User.where(token: token_from_header).take if token_valid?
```

Comment:
이런경우 take 대신 바로 find_by를 사용해주세요.
```suggestion
    current_user = User.find_by(token: token_from_header)if token_valid?
```

Reply:
[6ab6544](https://github.com/travelingaries/taxi-dispatch-backend-api/pull/1/commits/6ab6544dc200974330b37ac59f3005631008cbf6)에서 수정했습니다 🙆‍♂️